src/components/Items/Scorecard/Scorecard.vue

```diff
@@ -0,0 +1,252 @@
+          :key="reRenderKey"
+        >
+        </CircularProgress>
+
+        <!-- metric boxes -->
+        <div class="flex flex-row justify-between gap-4">
+          <div
+            v-for="metric in metrics"
+            class="w-full rounded-2xl bg-pastel-yellow grid grid-rows-2 lg:grid-rows-3 border-2"
+            :key="metric"
+          >
+            <div class="lg:row-span-2 w-full h-full flex flex-row justify-center gap-2">
+              <inline-svg
+                :src="getIconSource(metric.icon.source)"
+                class="h-4 md:h-6 lg:h-10 w-4 md:w-6 lg:w-10 place-self-center"
+                :class="metric.icon.color"
+              ></inline-svg>
+              <p class="text-base md:text-2xl lg:text-4xl font-bold my-auto">
+                {{ metric.value }}
+              </p>
+            </div>
+            <div class="text-center text-sm md:text-base lg:text-xl font-medium">
+              {{ metric.description }}
+            </div>
+          </div>
+        </div>
+
+        <!-- action buttons -->
+        <div class="place-self-center mt-3">
+          <icon-button
+            :titleConfig="watchAgainButtonTitleConfig"
+            :buttonClass="watchAgainButtonClass"
+            @click="watchAgain"
+            data-test="watchAgainButton"
+          ></icon-button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import CircularProgress from "@/components/UI/Progress/CircularProgress.vue";
+import Utilties from "@/services/Functional/Utilities.js";
+import IconButton from "@/components/UI/Buttons/IconButton.vue";
+
+const confetti = require("canvas-confetti");
+const confettiCanvas = document.getElementById("confetticanvas");
+const confettiHandler = confetti.create(confettiCanvas, { resize: true });
+
+export default {
+  name: "Scorecard",
+  components: {
+    CircularProgress,
+    IconButton,
+  },
+  props: {
+    metrics: {
+      // details of all the metrics to show
+      // format of one metric --
+      // {
+      //   name: "METRIC_NAME",
+      //   icon: {
+      //     source: "SVG_NAME.svg",
+      //     color: "TAILWIND_CLASSES",
+      //   },
+      //   value: VALUE_TO_SHOW,
+      //   description: DESCRIPTION_TO_SHOW,
+      // },
+      default: () => [],
+      type: Array,
+    },
+    progressPercentage: {
+      // progress to show on the progress bar in %
+      default: 0,
+      type: Number,
+    },
+    greeting: {
+      // greeting of the scorecard
+      default: "",
+      type: String,
+    },
+    showScorecard: {
+      // indicator of when the scorecard has popped up
+      default: false,
+      type: Boolean,
+    },
+    plioTitle: {
+      // plio's title
+      default: "",
+      type: String,
+    },
+    isPortrait: {
+      default: false,
+      type: Boolean,
+    },
+  },
+  data() {
+    return {
+      localProgressPercentage: 0, // local value of progress
+      showConfetti: false, // to show confetti or not
```

Comment:
based on earlier explanation
```suggestion
      isConfettiShown: false, // to show confetti or not
```